libkpod/stats.go

```diff
@@ -0,0 +1,176 @@
+package libkpod
+
+import (
+	"path/filepath"
+	"syscall"
+	"time"
+
+	"strings"
+
+	"github.com/opencontainers/runc/libcontainer"
+)
+
+// ContainerStats contains the statistics information for a running container
+type ContainerStats struct {
+	Container   string
+	CPU         float64
+	cpuNano     uint64
+	systemNano  uint64
+	MemUsage    uint64
+	MemLimit    uint64
+	MemPerc     float64
+	NetInput    uint64
+	NetOutput   uint64
+	BlockInput  uint64
+	BlockOutput uint64
+	PIDs        uint64
+}
+
+// GetContainerStatsOnce gets the running stats for a container one time and returns the stats
+func (c *ContainerServer) GetContainerStatsOnce(ids []string, all bool) ([]*ContainerStats, error) {
+	containerIDs, err := c.ListContainersFiltered(ids, all)
+	if err != nil {
+		return nil, err
+	}
+	containerStats := []*ContainerStats{}
+	for _, id := range containerIDs {
+		initialStats, err := c.getContainerStats(id, 0, 0)
+		if err != nil {
+			return nil, err
+		}
+		realStats, err := c.getContainerStats(id, initialStats.cpuNano, initialStats.systemNano)
+		if err != nil {
+			return containerStats, err
+		}
+		containerStats = append(containerStats, realStats)
+	}
+	return containerStats, nil
+}
+
+// GetContainerStatsFollow gets the running stats for a running container continually
```

Comment:
I also don't feel too strongly either way but I already implemented it, so I'm going to just leave it.   That being said, it seems to be implemented a bit awkwardly right now, so if either of you has input on how to clean that up it would be much appreciated